### PR TITLE
[APPSEC-12254] Update default API security sensitive data scanners

### PR DESCRIPTION
**Pull request by jkirsteins**
### What does this PR do?

Adding new default sensitive data scanners.

<details>
  <summary>Test values for waf_runner</summary>

  ```
{
    "server.request.headers.no_cookies": {
        "uk_nin_space": "QQ 123456 C",
        "uk_nin_nodelim": "QQ123456C",
        "iban": "FR7630006000011234567890189",
        "some_payment_account": "GI04BARC000001234567890",
        "SIN": "123-345-567",
        "Authorization": "Bearer 123542123",
        "WWW-Authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c",
        "Non-standard-JWT": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c",
        "email_no_tld": "test@example",
        "some address": "test@example.com",
        "australian_bankcard": "5610591081018250",
        
        "diners_4_6_4": "3056 930902 5904",
        "diners_4_4_4_2": "3056 9309 0259 04",
        "diners_8_6": "30569309 025904",
        "diners_1x14": "30569309025904",

        "amex_4_6_5": "3782 822463 10005",
        "amex_4_4_4_3": "3782 8224 6310 005",
        "amex_4_5_6": "3714 49635 398431",
        "amex_8_7": "37144963 5398431",
        "amex_1x15": "371449635398431",
        "amex_corporate": "3787 34493 671000",
        
        "mastercard_1x16": "5555555555554444",
        "mastercard_4x4": "5555 5555 5555 4444",
        "mastercard_2x8": "55555555 55554444",
        
        "discover_1x16": "6011111111111117",
        "discover_2x8": "60111111 11111117",
        "discover_4x4": "6011 1111 1111 1117",

        "jcb_1x16": "3579666906613115",
        "jcb_2x8": "35792387 52209248",
        "jcb_4x4": "3588 8218 6900 8535",

        "maestro_1x16": "6765232279738554",
        "maestro_4x4": "6758 6286 7856 5739",
        "maestro_2x8": "50280417 49024627",

        "visa_short": "4222222222222",
        "visa_4x4": "4012 8888 8888 1881",
        "visa_2x8": "40128888 88881881",
        "visa_1x16": "4012888888881881",

        "ssn_space": "000 11 1111",
        "ssn_dash": "457-55-5462",
        "ssn_dot": "457.55.5462",
        "ssn_without_delim_should_not": "457555462",

        "dankort": "76009244561",
        "switch_solo": "6331101999990016",
        "e-mail": "test@example.com"
    },
    "waf.context.processor": {
        "extract-schema": true
    }
}
  ```
</details>

### Checklist
- [ ] Did you added the `test_vectors` for your rules
- [x] Did you tested it on your local machine first
- [ ] Did you tested it on staging

### CODE REVIEW

File: scanners/iban.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+id: 8VS2RKxzR8a_95L5fuwaXQ
+name: IBAN
+# key:
```
Comment (Anilm3) [inline scanners/iban.yaml:3]: Was this supposed to stay commented?

File: processors/extract-content.yaml [hunk 2]
```diff
@@ -27,6 +27,8 @@ parameters:
         - address: server.response.body
       output: _dd.appsec.s.res.body
   scanners:
+    - tags:
+        category: payment
```
Comment (Anilm3) [inline processors/extract-content.yaml:31]: There's an `extract-headers.yaml`, would it be relevant to have this new category there? (maybe not but just checking)

**Review by Anilm3 — COMMENTED** [review]
Without going into the details of each scanner, the main reason we have keys on most of them is to avoid false positives. I suggest you consider a known set of keys for all credit cards as the patterns can be quite common in arbitrary data.

File: scanners/iban.yaml (hunk 1 above)
Comment (jkirsteins) [reply scanners/iban.yaml]: Missed this. Re-added the key.

**Comment by jkirsteins** [comment]
> Without going into the details of each scanner, the main reason we have keys on most of them is to avoid false positives. I suggest you consider a known set of keys for all credit cards as the patterns can be quite common in arbitrary data.

Thanks, that's a good point. I went over it, and re-added keys to the cards. The only one I left without a key was the JWT one.

File: processors/extract-content.yaml (hunk 2 above)
Comment (Taiki-San) [reply processors/extract-content.yaml:31]: I don't really see a case for sending credit cards in headers. Probably not worth the CPU cycles it'll take to run them

File: scanners/email.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  operator: match_regex
+  parameters:
+    regex: mail|email|address
+    options:
+      case_sensitive: false
+      min_length: 4
+value:
+  operator: match_regex
+  parameters:
+    regex: \b[\w!#$%&'*+/=?`{|}~^-]+(?:\.[\w!#$%&'*+/=?`{|}~^-]+)*(%40|@)(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,6}\b
```
Comment (Taiki-San) [inline scanners/email.yaml:13]: The regex is incomplete, it won't catch IPs in the domain (`root@127.0.0.1`). Not a massive deal however

File: scanners/uk_national_insurance_number.yaml [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-    regex: national[\s_]?(?:insurance(?:\s+number)?)?|NIN|NI[\s_]?number|insurance[\s_]?number
+    regex: nin|insurance
     options:
       case_sensitive: false
       min_length: 3
 value:
   operator: match_regex
   parameters:
-    regex: \b[A-Z]{2}\d{6}[A-Z]?\b
+    regex: \b[A-Z]{2}[\s-]{0,1}\d{6}[\s-]{0,1}[A-Z]?\b
```
Comment (Taiki-San) [inline scanners/uk_national_insurance_number.yaml:13]:
```suggestion
    regex: \b[A-Z]{2}[\s-]?\d{6}[\s-]?[A-Z]?\b
```

**Review by Anilm3 — APPROVED** [review]

File: scanners/email.yaml (hunk 3 above)
Comment (jkirsteins) [reply scanners/email.yaml:13]: Discussed async w @Taiki-San , keeping this e-mail for now for consistency with other places where this regex is used.

**Comment by jkirsteins** [comment]
The failing system tests check is due to an old Go version. Not related to this PR, confirmed with @cbeauchesne that it's fine to skip.

**Pull request merged**